python/lsst/ip/isr/vignette.py

```diff
@@ -52,39 +53,136 @@ class VignetteConfig(Config):
         doc="Number of points used to define the vignette polygon.",
         default=100,
     )
-    doWriteVignettePolygon = Field(
-        dtype=bool,
-        doc="Persist polygon used to define vignetted region?",
-        default=False,
-    )
 
 
 class VignetteTask(Task):
-    """Define a simple circular vignette pattern.
+    """Define a simple circular vignette pattern and optionally update mask
+    plane.
     """
     ConfigClass = VignetteConfig
     _DefaultName = "isrVignette"
 
-    def run(self, exposure):
+    def run(self, exposure=None, doUpdateMask=True, maskPlane="NO_DATA", vignetteValue=None, log=None):
         """Generate circular vignette pattern.
 
         Parameters
         ----------
         exposure : `lsst.afw.image.Exposure`
-            Exposure to construct vignette for.
+            Exposure to construct, apply, and optionally mask vignette for.
 
         Returns
         -------
-        polygon : `lsst.afw.Geom.Polygon`
+        polygon : `lsst.afw.geom.Polygon`
             Polygon defining the boundary of the vignetted region.
         """
+        theta = np.linspace(0, 2*np.pi, num=self.config.numPolygonPoints, endpoint=False)
+        x = self.config.radius*np.cos(theta) + self.config.xCenter
+        y = self.config.radius*np.sin(theta) + self.config.yCenter
+        points = np.array([x, y]).transpose()
+        fpPolygon = afwGeom.Polygon([geom.Point2D(x1, y1) for x1, y1 in reversed(points)])
+        if exposure is None:
+            return fpPolygon
+
+        # Exposure was provided, so attach the validPolygon associated with the
+        # vignetted region.
+        setValidPolygonCcdIntersect(exposure, fpPolygon, log=log)
+
+        if doUpdateMask:
+            polygon = exposure.getInfo().getValidPolygon()
+            maskVignettedRegion(exposure, polygon, maskPlane="NO_DATA", vignetteValue=vignetteValue, log=log)
+        return fpPolygon
+
+
+def setValidPolygonCcdIntersect(ccdExposure, fpPolygon, log=None):
+    """Set valid polygon on ccdExposure associated with focal plane polygon.
+
+    Where the ccd exposure's valid polygon is considered the intersection of
```

Comment:
Maybe rephrase a bit with something like: "The CCD exposure's valid polygon is defined as the intersection of ... "